src-unix/native/winsize.c

```diff
@@ -10,7 +10,16 @@
 CAMLprim value caml_notty_winsize (value vfd) {
 #ifdef _WIN32
   (void) vfd;
-  caml_failwith("not implemented on windows");
+  HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+  if (hConsole == INVALID_HANDLE_VALUE) return Val_int (0);
```

Comment:
Here is the Windows API doc for this by the way: https://learn.microsoft.com/en-us/windows/console/getstdhandle#return-value

Reply:
Shouldn't you check that the output is really a Console, and not something else (eg a file in case of redirection)?

Reply:
Maybe `GetConsoleScreenBufferInfo` fails if the handle isn't a console. We could try in chain stdout, stderr, stdin, in case one of the fd is redirected.

Reply:
> The [CreateFile](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea) function enables a process to get a handle to its console's input buffer and active screen buffer, even if `STDIN` and `STDOUT` have been redirected. To open a handle to a console's input buffer, specify the `CONIN$` value in a call to `CreateFile`. Specify the `CONOUT$` value in a call to `CreateFile` to open a handle to a console's active screen buffer. `CreateFile` enables you to specify the read/write access of the handle that it returns.

From the [Console Handles](https://learn.microsoft.com/en-us/windows/console/console-handles) doc page. Maybe that's the way forward.